server/channels/api4/post_utils.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func postPermissionCheck(c *Context, channelId string) {
```

Comment:
Could these methods be renamed to better explain what they're checking? I'm thinking of things like `Context.RequireBotUserId` in `channels/web/context.go` even though these methods do a bit more than those

Reply:
What they are checking can change, and the idea of these functions is, for example with `postPermissionCheck`, any permission check needed for creating a post goes in here. Similar for hardened mode and and post priority, I think the generic names are fine. Feel free to make a recommendation of a new name and I'll rename all those.